---
title: Data Visualization

---

**NBIS workshop on data visualization in R**
# Data Visualization 

Welcome to the *Workshop on Data Visualization* organised by National Bioinformatics Infrastructure, Sweden. 

This document is a life record of what we are doing and learning during the workshop. As either a student or a teacher, you are encouraged to write your comments, questions and feedback.

![](https://i.redd.it/pr02mjbz8dsy.png)

## Useful resources

- **Course website**: https://uppsala.instructure.com/courses/82719
- **HackMD (this document):** https://hackmd.io/BcwXnvdVS469hIHXnDSvJg?both 
- **Solution to exercise:** https://uppsala.instructure.com/courses/82719/pages/solution-link?module_item_id=752627

# Beginners

- **Basic R**: https://immuntech.github.io/Bioinformatics_Workshop_ImmTech/Data/2019-10-15/R_exercise_Lokesh.html

## Data Download

https://nbisweden.github.io/workshop-data-visualization-r/2304/data/data.zip

## Introduction
Here, we write a few words about ourselves to get to know each other a bit better.

#### Teachers
* **Lokeshwaran Manoharan** - I am the course leader for this workshop. I work as a Bioinformatics expert at NBIS in the medical faculty node of Lund University. My research interest lies very much in the genomics of non-model organisms and microbes in particular.

* **Markus Ringnér** - I am a teacher for this workshop. I am based in Lund under the Department of Biology. I work as a bioinformatician in the NBIS long term support.

* **Yuan Li** - I am a TA for this workshop. I am based in Lund under the Department of Immunotechnology. I work as a bioinformatician in the NBIS short term support. My field lies broadly in transcriptomics.

#### Participants



## Some important links

* [**Markdown syntax**](https://www.markdownguide.org/basic-syntax/)
* [**ggplot2 cheatsheet**](https://www.maths.usyd.edu.au/u/UG/SM/STAT3022/r/current/Misc/data-visualization-2.1.pdf)
* [**A really good and comprehensive R heatmap package**](https://jokergoo.github.io/ComplexHeatmap-reference/book/index.html) - Complex Heatmap
* [**R graph Gallery**](https://www.r-graph-gallery.com/)
* [**RShiny cheatsheet**](https://shiny.rstudio.com/images/shiny-cheatsheet.pdf)
* [**Statistics layers on ggplot2**](http://www.sthda.com/english/articles/24-ggpubr-publication-ready-plots/76-add-p-values-and-significance-levels-to-ggplots/)

## Day 1

### Questions & Issues


- [ ] Example: **What does the parameter `by = 0` mean in `merged_data_wide <- merge(md, t(gc), by = 0)` ?**  

> **Answer:** Here, the `merge` function will combine two data frames by their rownames, i.e., by invoking the parameter `by = 0`.  Note that in the exercise, we have explicitly assigned the rownames of the dataframe by executing command `rownames(md) <- md$Sample_ID`.



### Bugs


### Feedback

- [ ] Suggestion: Add information regarding the hjust argument somewhere in proximity to 3.1 Axis style! And thanks for the lunch and weather by the way! /AxelC
- [ ] 



## Day 2

### Questions & Issues



### Bugs



### Feedback




## Day 3

### Questions & Issues


### Feedback



### Bugs

**NBIS workshop on data visualization in R**
# Data Visualization 

Welcome to the *Workshop on Data Visualization* organised by National Bioinformatics Infrastructure, Sweden. 

This document is a life record of what we are doing and learning during the workshop. As either a student or a teacher, you are encouraged to write your comments, questions and feedback.

![](https://i.redd.it/pr02mjbz8dsy.png)

## Useful resources

- **Course website**: https://uppsala.instructure.com/courses/82719
- **HackMD (this document):** https://hackmd.io/BcwXnvdVS469hIHXnDSvJg?both 
- **Solution to exercise:** https://uppsala.instructure.com/courses/82719/pages/solution-link?module_item_id=752627

# Beginners

- **Basic R**: https://immuntech.github.io/Bioinformatics_Workshop_ImmTech/Data/2019-10-15/R_exercise_Lokesh.html

## Data Download

https://nbisweden.github.io/workshop-data-visualization-r/2304/data/data.zip

## Introduction
Here, we write a few words about ourselves to get to know each other a bit better.

#### Teachers
* **Lokeshwaran Manoharan** - I am the course leader for this workshop. I work as a Bioinformatics expert at NBIS in the medical faculty node of Lund University. My research interest lies very much in the genomics of non-model organisms and microbes in particular.

* **Markus Ringnér** - I am a teacher for this workshop. I am based in Lund under the Department of Biology. I work as a bioinformatician in the NBIS long term support.

* **Yuan Li** - I am a TA for this workshop. I am based in Lund under the Department of Immunotechnology. I work as a bioinformatician in the NBIS short term support. My field lies broadly in transcriptomics.

#### Participants



## Some important links

* [**Markdown syntax**](https://www.markdownguide.org/basic-syntax/)
* [**ggplot2 cheatsheet**](https://www.maths.usyd.edu.au/u/UG/SM/STAT3022/r/current/Misc/data-visualization-2.1.pdf)
* [**A really good and comprehensive R heatmap package**](https://jokergoo.github.io/ComplexHeatmap-reference/book/index.html) - Complex Heatmap
* [**R graph Gallery**](https://www.r-graph-gallery.com/)
* [**RShiny cheatsheet**](https://shiny.rstudio.com/images/shiny-cheatsheet.pdf)
* [**Statistics layers on ggplot2**](http://www.sthda.com/english/articles/24-ggpubr-publication-ready-plots/76-add-p-values-and-significance-levels-to-ggplots/)

## Day 1

### Questions & Issues


- [ ] Example: **What does the parameter `by = 0` mean in `merged_data_wide <- merge(md, t(gc), by = 0)` ?**  

> **Answer:** Here, the `merge` function will combine two data frames by their rownames, i.e., by invoking the parameter `by = 0`.  Note that in the exercise, we have explicitly assigned the rownames of the dataframe by executing command `rownames(md) <- md$Sample_ID`.



### Bugs


### Feedback

- [ ] Suggestion: Add information regarding the hjust argument somewhere in proximity to 3.1 Axis style! And thanks for the lunch and weather by the way! /AxelC
- [ ] 



## Day 2

### Questions & Issues



### Bugs



### Feedback




## Day 3

### Questions & Issues


### Feedback



### Bugs

