**NBIS Workshop**
# Data Visualization in R Apr-2022

Welcome to the *Workshop on Data Visualization in R* workshop organised by National Bioinformatics Infrastructure, Sweden. 

This document is a life record of what we are doing and learning during the workshop. As either a student or a teacher, you are encouraged to write your comments, questions and feedback.

![](https://i.redd.it/pr02mjbz8dsy.png)

## Useful resources

- **Course website**: https://uppsala.instructure.com/courses/46547
- **HackMD (this document):** https://hackmd.io/BcwXnvdVS469hIHXnDSvJg?both 
- **Solution to exercise https://nbisweden.github.io/workshop-data-visualization-r/2204/home_content.html#6_Solutions_to_exercises:



## Data Download

https://nbisweden.github.io/workshop-data-visualization-r/2204/data/data.zip

## Introduction
Here, we write a few words about ourselves to get to know each other a bit better.

#### Teachers
* **Lokeshwaran Manoharan** - I am the course leader for this workshop. I work as a Bioinformatics expert at NBIS in the medical faculty node of Lund University. My research interest lies very much in the genomics of non-model organisms and microbes in particular.

* **Markus Ringnér** - I am one of the teachers for this workshop. I work in the NBIS long term support team and have my office at the Biology Department, Lund University. I have a background in cancer genomics.

* **Louella Vasquez** - I am one of the teachers for this workshop. I am based in Lund under the medical faculty.  My background is in epigenetics and genetic association QTL analysis.  I love cats.

* **Juliana Assis** - I am a TA for this workshop. I am based in Lund under the Department of Immunotechnology. I work as a bioinformatician in a short term support. My field lies broadly in multi-omics data and data integration.



#### Students
* **Eleftheria Theodoropoulou** - I am a PhD student (4th year) in Stockholm University and I am working within toxicology. In my group we are interested in the potential epigenetic and transcriptomic effects that the exposure to certain chemical compounds can have on tissues of humans and animal models in order to find mechanisms of toxicity. I am also very allergic to pollen so I will keep sneezing I am afraid :(

* **Hanna  Bensch** - I am a PhD student at Linnaeus university in Kalmar, did my masters here in Lund so fun to be back! In my PhD I study cooperative behaviour and gut mcrobiome of Damaraland mole-rats in the Kalahari Desert. 

* **Qing Fang** -I am a postdoc working in Copenhagen University. I work with gut microbiome(wetlab, metabolomics etc) and I am interested in learning more R skills for data analysis and visualization. I love all kinds of sports and specially running.

* **Kelin Oliveira** - I am a PhD student from Lund University. I am currently working with multi omics integration to search for putative therapeutical targets in glioblastoma tumors. The main fronts I work with are transcriptomics, proteomics and exome.

* **Fabricio Romero Garcia** - I'm a PhD student from Karolinska Institutet. My project is about the vaginal microbiome during healthy and diseased conditions. We want to know the mechanistics on the interaction between the microbiota and the host, and how this knowledge might be used for treating viral infections like HPV. The project covers long-term cohort studies as well, which supports the different states during the continuum of health and disease. I am also a proud alumni from Lund University :).

* **Arturo Ibáñez Fonseca** - I'm a postdoc at the Lung Biology group at Lund University. My background is on the use of biomaterials in regenerative medicine. My current project aims at elucidating how human lung cells interact with their surroundings by building defined ECM-like microenvironments *in vitro*. Hopefully transcriptomic analysis will help me in this endeavour :D

* **Sara Häggström** - PhD student from Umeå University and department of Plant Physiology. I work with Ribo-Seq data and am interested in translational regulation in plant cells. Work 50/50 in the lab and with bioinformatics.

* **Hongkai Zhang** - PhD student from Lund University. I do bird genomics and the bird species is Great reed warbler (*Acrocephalus arundinaceus*). Specifically, I study the genetic architecture of sexual conflict, using this species.

* **Franca Stábile** - I am a PhD student from Lund University at the department of Biology/Aquatic Ecology Unit. I am working with the effects of nano-plastics at freshwater ecosystems, using different end-points from macro-invertebrate populations to ecosystem level functioning.

* **Zoé Pochon** - I am a PhD student in scientific archaeology at the Centre for Palaeogenetics of Stockholm University. I look for pathogens in humans from different archaological sites. In the second phase of my PhD I will try to look for RNA pathogens as well if we manage to retrieve RNA from the ancient bones. 

## Some important links

* [**Markdown syntax**](https://www.markdownguide.org/basic-syntax/)
* [**ggplot2 cheatsheet**](https://www.maths.usyd.edu.au/u/UG/SM/STAT3022/r/current/Misc/data-visualization-2.1.pdf)
* [**A really good and comprehensive R heatmap package**](https://jokergoo.github.io/ComplexHeatmap-reference/book/index.html) - Complex Heatmap
* [**R graph Gallery**](https://www.r-graph-gallery.com/)
* [**RShiny cheatsheet**](https://shiny.rstudio.com/images/shiny-cheatsheet.pdf)

## Wednesday - Day 1

### Questions & Issues


- [ ] Example: **What does the parameter `by = 0` mean in `merged_data_wide <- merge(md, t(gc), by = 0)` ?**  

> **Answer:** Here, the `merge` function will combine two data frames by their rownames, i.e., by invoking the parameter `by = 0`.  Note that in the exercise, we have explicitly assigned the rownames of the dataframe by executing command `rownames(md) <- md$Sample_ID`.

- [ ] **When I am plotting a barplot using ggplot I understand that I can use `geom_col()` as suggested in the markdown. However, when plotting my own graphs using ggplot I often use `geom_bar`. In doing so I have to speicify `stat = "identity"` in the function. I mainly took over this codes from other colleagues in my lab and never questioned it. Now I however wonder why I have to specify it and what exactly it means. Maybe you can explain the differnces between `geom_col()` and `geom_bar()` and why I have to specify `stat` ?**


> You can view `geom_bar()` as having more functionality than `geom_col()`. If one has data that are paired as in group/x-value and a count/y-value for each group/x-value, one can generate a barplot for this data using `geom_col()`. `geom_bar()`on the other hand does not expect a y-value (count in this case), instead it calculates the y-value from the data (from the x-values). For example with `stat="count"` `geom_bar()`sums up the number of times each group/x-value appears and makes a barplot of that, but `geom_bar()` can also aggregate the data in more customizable ways. If one already has the number of times each group appears then `geom_col()` is a more simple notation to generate columns in a barplot, but `geom_bar()`provides this same functionality with `stat="identity` since it then overides the aggregation that `geom_bar()`can do in different ways. 

- [ ] **Are there always as many PCs as samples (observations?)** 

> Technically yes! But this is because PCA just ries to compartmentalize  the variation from each sample into components! Maybe the first few components explain most of the variation 


### Bugs



### Feedback

* `pivot_longer` instead of `gather` could be the right one to use for the future




## Thursday - Day 2

### Questions & Issues



### Bugs



### Feedback




## Friday - Day 3

### Questions & Issues


### Feedback
* We can add to the ui:

```
conditionalPanel(condition = "input.in_duration_date_start < input.in_duration_date_end",
                         textOutput('error_duration2')),
```
and to the server:
```
output$error_duration2 <- renderText({
      shiny::validate(
        shiny::need(input$in_duration_date_start < input$in_duration_date_end, 
                    'The end date must be after the start date'
        )
      )
    })
```
to add the error message about putting the dates in the wrong order.


### Bugs
* Updating `dplyr`, `knitr`and `RSQlite` fixed the problem I had with error not being substituted or hidden when I used `validate`and `need`. I guess `dplyr`had something to do with it?

* Try this instead in the section **5 Error validation** as an example of a column that we cannot make a histogram of. The current example was not giving us a choice to select an "incorrect column".


```
shinyApp(
  ui=fluidPage(
    selectInput("data_input",label="Select data",choices=c("mtcars","faithful","iris")),
    selectInput("header_input",label="Select column name",choices=NULL),
    plotOutput("plot_output",width="400px")
  ),
  server=function(input,output,session) {
    getdata <- reactive({ get(input$data_input, 'package:datasets') })

    
    observe({
      updateSelectInput(session,"header_input",label="Select column name",choices=colnames(getdata()))
    })
    
    output$plot_output <- renderPlot({
      shiny::req(input$header_input)
      validate(need(is.numeric(getdata()[, input$header_input]), message="Is not numeric"))
      hist(getdata()[, input$header_input],xlab=input$header_input,main=input$data_input)
    })
  },
  options=list(height=600))
```

* **Section "5. Error validation"** (second chunk). A way to get custom error messages both when the `data_input` is empty and when it contains a string that does not correspond to the name of any data set:

```
getdata <- reactive({
validate(need(try(input$data_input),"Please select a data set"))
validate(need(exists(input$data_input),"Please select an existing data set"))
get(input$data_input,'package:datasets')

})
```

## Feedback during feedback session

**How to design optimal plots?**

* Books by [Edward R Tufte](https://www.edwardtufte.com/tufte/). The original trilogy starting with "The Visual Display of Quantitative Information", and then onwards to the most recent book "Seeing with Fresh Eyes: Meaning, Space, Data, Truth".

* Useful links for plot inspiration:
    * https://r-graph-gallery.com/index.html
    * https://www.data-to-viz.com/
