---
title: Data Visualization

---

**NBIS workshop on Advanced Data Visualization**
# Data Visualization 

Welcome to the *Workshop on Advanced Data Visualization* organised by National Bioinformatics Infrastructure, Sweden. 

This document is a life record of what we are doing and learning during the workshop. As either a student or a teacher, you are encouraged to write your comments, questions and feedback.

![](https://i.redd.it/pr02mjbz8dsy.png)

## Useful resources

- **Course website**: https://nbisweden.github.io/workshop-data-visualization-r/2505/
- **HackMD (this document):** https://hackmd.io/BcwXnvdVS469hIHXnDSvJg?both 

# Beginners

- **Basic R**: https://immuntech.github.io/Bioinformatics_Workshop_ImmTech/Data/2019-10-15/R_exercise_Lokesh.html

## Data Download

## Introduction
Here, we write a few words about ourselves to get to know each other a bit better.

#### Teachers
* **Lokeshwaran Manoharan** - I am the course leader for this workshop. I work as a Bioinformatics expert at NBIS in the medical faculty node of Lund University. My research interest lies very much in the genomics of non-model organisms and microbes in particular.

* **Markus Ringnér** - I am a teacher for this workshop. I am based in Lund under the Department of Biology. I work as a bioinformatician at NBIS with projects in the peer review support track.

* **Katja Kozjek** - I am a teacher for this workshop. I work as a bioinformatician at NBIS in the Medical faculty node of Lund University. I am mainly working with metagenomics data; soil, fungal, human microbiomes.

*  **Louella Vasquez** -

* **Massimiliano Volpe** - I am a teacher for this workshop. I work as a bioinformatician at NBIS in the Immunotechnology Department (Lund University). I work with transcriptomics, WGS and WES data.

#### Participants
* **Zeeshan Syed** (ZAS) - I am a postdoc at the Biology department at Linköping University. I am curious about finding out the genetic mechinasms underpinning how natural and sexual selection interact to shape the evolution of sex-specific lifespan and aging, from both macro- and microevolutionary perspectives. I work with phylogenetic/genomic, transcriptomic and demographic data. 


## Some important links

* [**Markdown syntax**](https://www.markdownguide.org/basic-syntax/)
* [**ggplot2 cheatsheet**](https://rstudio.github.io/cheatsheets/data-visualization.pdf)
* [**A really good and comprehensive R heatmap package**](https://jokergoo.github.io/ComplexHeatmap-reference/book/index.html) - Complex Heatmap
* [**R graph Gallery**](https://www.r-graph-gallery.com/)
* [**RShiny cheatsheet**](https://raw.githubusercontent.com/rstudio/cheatsheets/main/shiny.pdf)
* [**Statistics layers on ggplot2**](http://www.sthda.com/english/articles/24-ggpubr-publication-ready-plots/76-add-p-values-and-significance-levels-to-ggplots/)

## Day 1

### Questions & Issues


- [ ] Example: **What does the parameter `by = 0` mean in `merged_data_wide <- merge(md, t(gc), by = 0)` ?**  

> **Answer:** Here, the `merge` function will combine two data frames by their rownames, i.e., by invoking the parameter `by = 0`.  Note that in the exercise, we have explicitly assigned the rownames of the dataframe by executing command `rownames(md) <- md$Sample_ID`.

ZAS - Is it okay to share the course material with interested labmates that did/could not attend the workshop? 

K: Yes, course material is publically available and you are welcome to share it with your lab/group. 


### Bugs


### Feedback

ZAS - Could you please provide downloadable files for the materials? 

## Day 2

### Questions & Issues

ZAS - Is there a way for importing citations in a qmd? If yes, is the bibliography formatting standard (e.g., APA), or can we make journal specific formating?

K: Yes, you can import citations to **.qmd** file. In short, you need to have **.bib** file where you stored you references, i.e. BibTeX entries. You format BibTeX files using a reference manager.

An example of BibTeX entry is: 

```
@book{Hemingway1952,
  title={The Old Man and the Sea},
  author={Hemingway, Ernest},
  year={1952},
  publisher={Charles Scribner's Sons}
}
```

**Entry-type:** with *@book* we define the type according to the schema *@type* of the reference. Possible are *@article* for scientific articles and others. BibTeX likes to specify which fields are optional and which are required to indicate them correctly in the literature.

**Entry fields:** in this case of our *@book* example, these are title, author, year and publisher. 

**citation-key:** in our example, it is Hemingway1952 and is used to indicate an in-text citation in LaTeX, i.e., to refer to the source. In Quarto we do this with [@Hemingway1952]. The citation key can be any string - often a combination of author, year, and a word from the title.

In the YAML you then specify **.bib file**. 

```
---
title: 
author:
output: html_document
bibliography: references.bib
---
```

Once you have this file ready you can change it to a specific journal format using [citation style language](https://citationstyles.org/). 

```
---
title:
author:
output: html_document
bibliography: references.bib
csl: my_journal.csl
---
```

When you cite articles in Quarto document you use [@article-handle] at the location where you want the in-text citation to appear, in our case we would uses [@Hemingway1952]

More info [here](https://quarto.org/docs/authoring/citations.html) 

**PCA in 3D** 

You can use plotly and scatter3d
https://plotly.com/r/pca-visualization/

https://jackauty.com/pca-and-3d-pca/

https://bryanhanson.github.io/LearnPCA/articles/Vig_05_Visualizing_PCA_3D.html

https://www.sthda.com/english/wiki/a-complete-guide-to-3d-visualization-device-system-in-r-r-software-and-data-visualization#d-scatter-plot-1


### Bugs



### Feedback

ZAS - Amazing day, really enjoyed the exercises, and very gald to have moved over to Quarto. And of course, thanks for the dinner. 


## Day 3

### Questions & Issues


### Feedback

ZAS - Sharing a neat package I came across recently: [pacman](https://trinker.github.io/pacman/vignettes/Introduction_to_pacman.html).
It provides neat ways to install and load packages imo. 

### Bugs

**NBIS workshop on Advanced Data Visualization**
# Data Visualization 

Welcome to the *Workshop on Advanced Data Visualization* organised by National Bioinformatics Infrastructure, Sweden. 

This document is a life record of what we are doing and learning during the workshop. As either a student or a teacher, you are encouraged to write your comments, questions and feedback.

![](https://i.redd.it/pr02mjbz8dsy.png)

## Useful resources

- **Course website**: https://nbisweden.github.io/workshop-data-visualization-r/2505/
- **HackMD (this document):** https://hackmd.io/BcwXnvdVS469hIHXnDSvJg?both 

# Beginners

- **Basic R**: https://immuntech.github.io/Bioinformatics_Workshop_ImmTech/Data/2019-10-15/R_exercise_Lokesh.html

## Data Download

## Introduction
Here, we write a few words about ourselves to get to know each other a bit better.

#### Teachers
* **Lokeshwaran Manoharan** - I am the course leader for this workshop. I work as a Bioinformatics expert at NBIS in the medical faculty node of Lund University. My research interest lies very much in the genomics of non-model organisms and microbes in particular.

* **Markus Ringnér** - I am a teacher for this workshop. I am based in Lund under the Department of Biology. I work as a bioinformatician at NBIS with projects in the peer review support track.

* **Katja Kozjek** - I am a teacher for this workshop. I work as a bioinformatician at NBIS in the Medical faculty node of Lund University. I am mainly working with metagenomics data; soil, fungal, human microbiomes.

*  **Louella Vasquez** -

* **Massimiliano Volpe** - I am a teacher for this workshop. I work as a bioinformatician at NBIS in the Immunotechnology Department (Lund University). I work with transcriptomics, WGS and WES data.

#### Participants
* **Zeeshan Syed** (ZAS) - I am a postdoc at the Biology department at Linköping University. I am curious about finding out the genetic mechinasms underpinning how natural and sexual selection interact to shape the evolution of sex-specific lifespan and aging, from both macro- and microevolutionary perspectives. I work with phylogenetic/genomic, transcriptomic and demographic data. 


## Some important links

* [**Markdown syntax**](https://www.markdownguide.org/basic-syntax/)
* [**ggplot2 cheatsheet**](https://rstudio.github.io/cheatsheets/data-visualization.pdf)
* [**A really good and comprehensive R heatmap package**](https://jokergoo.github.io/ComplexHeatmap-reference/book/index.html) - Complex Heatmap
* [**R graph Gallery**](https://www.r-graph-gallery.com/)
* [**RShiny cheatsheet**](https://raw.githubusercontent.com/rstudio/cheatsheets/main/shiny.pdf)
* [**Statistics layers on ggplot2**](http://www.sthda.com/english/articles/24-ggpubr-publication-ready-plots/76-add-p-values-and-significance-levels-to-ggplots/)

## Day 1

### Questions & Issues


- [ ] Example: **What does the parameter `by = 0` mean in `merged_data_wide <- merge(md, t(gc), by = 0)` ?**  

> **Answer:** Here, the `merge` function will combine two data frames by their rownames, i.e., by invoking the parameter `by = 0`.  Note that in the exercise, we have explicitly assigned the rownames of the dataframe by executing command `rownames(md) <- md$Sample_ID`.

ZAS - Is it okay to share the course material with interested labmates that did/could not attend the workshop? 

K: Yes, course material is publically available and you are welcome to share it with your lab/group. 


### Bugs


### Feedback

ZAS - Could you please provide downloadable files for the materials? 

## Day 2

### Questions & Issues

ZAS - Is there a way for importing citations in a qmd? If yes, is the bibliography formatting standard (e.g., APA), or can we make journal specific formating?

K: Yes, you can import citations to **.qmd** file. In short, you need to have **.bib** file where you stored you references, i.e. BibTeX entries. You format BibTeX files using a reference manager.

An example of BibTeX entry is: 

```
@book{Hemingway1952,
  title={The Old Man and the Sea},
  author={Hemingway, Ernest},
  year={1952},
  publisher={Charles Scribner's Sons}
}
```

**Entry-type:** with *@book* we define the type according to the schema *@type* of the reference. Possible are *@article* for scientific articles and others. BibTeX likes to specify which fields are optional and which are required to indicate them correctly in the literature.

**Entry fields:** in this case of our *@book* example, these are title, author, year and publisher. 

**citation-key:** in our example, it is Hemingway1952 and is used to indicate an in-text citation in LaTeX, i.e., to refer to the source. In Quarto we do this with [@Hemingway1952]. The citation key can be any string - often a combination of author, year, and a word from the title.

In the YAML you then specify **.bib file**. 

```
---
title: 
author:
output: html_document
bibliography: references.bib
---
```

Once you have this file ready you can change it to a specific journal format using [citation style language](https://citationstyles.org/). 

```
---
title:
author:
output: html_document
bibliography: references.bib
csl: my_journal.csl
---
```

When you cite articles in Quarto document you use [@article-handle] at the location where you want the in-text citation to appear, in our case we would uses [@Hemingway1952]

More info [here](https://quarto.org/docs/authoring/citations.html) 

**PCA in 3D** 

You can use plotly and scatter3d
https://plotly.com/r/pca-visualization/

https://jackauty.com/pca-and-3d-pca/

https://bryanhanson.github.io/LearnPCA/articles/Vig_05_Visualizing_PCA_3D.html

https://www.sthda.com/english/wiki/a-complete-guide-to-3d-visualization-device-system-in-r-r-software-and-data-visualization#d-scatter-plot-1


### Bugs



### Feedback

ZAS - Amazing day, really enjoyed the exercises, and very gald to have moved over to Quarto. And of course, thanks for the dinner. 


## Day 3

### Questions & Issues


### Feedback

ZAS - Sharing a neat package I came across recently: [pacman](https://trinker.github.io/pacman/vignettes/Introduction_to_pacman.html).
It provides neat ways to install and load packages imo. 

### Bugs

